### DISCUSSION TITLE
Testing with pre recorded samples

### DISCUSSION THREAD

**Discussion by CodedByMistake** [mcguirepr89/BirdNET-Pi#1067] [Q&A] [answered]
Hi everyone,
I'm a Computer Science student doing my last year project about identifing bird species in a farming field and I'm currently experimenting with Birdnet Pi on a Raspberry pi 4B 4GB and would like to know if its possible to upload an audio sample with a specific species of bird and for it to be processed and if it is how is it done?
I have tried to upload using the file manager but it says todays folder isn't writable so if there is a way I would really apreciate the help.
We are thinking on a raspberry pi 4 as a central node and having other devices capture and send it to the central node for processing but also using various raspberrys pi zero 2W as nodes doing the processing on their own and, if possible, using a central database that saves everything. Is is possible to have multiple devices running Birdnet Pi sepratly and then have all the date be joined and possibly identify which device got those results?
Thank you in advance,
Fábio

**Comment by tphakala** [comment, marked as answer]
I don't think that BirdNET-Pi is capable of doing any of those things you ask for. I am working on app which can do on-demand file analysis and has ability to extend to multi node setup you ask for, my app just doesn't have any web ui at this point, it is CLI only. I am working on web ui though. Please take a look at https://github.com/tphakala/go-birdnet and share your thoughts.

**Reply by CodedByMistake** [reply]
Hi, I actually saw your project when searching for something that used birdnet but was trying out birdnet pi.
I don’t use GitHub a lot but is there a way for us to chat privately ? If your project has the ability to do so I would love to try it out and be in contact with you to know how it works

**Reply by tphakala** [reply]
You can contact me by email at tomi.hakala (at) pobox.fi. Multi-node deployment model is not yet implemented, but it is not difficult to do, just takes some time.

**Reply by CodedByMistake** [reply]
Thank you very much for the contact information and availability. I will try it out ASAP and will contact you for help or information.
